### ISSUE TITLE
Nested parallel states don’t bubble up the `done` event

### ISSUE THREAD

**Issue by johnnydecimal** [davidkpiano/xstate#2349]
Nested parallel states don’t seem to bubble up the `done` event. As a result, the `onDone` transition is not taken.

Both the docs and the spec seem to indicate that this should be possible. 

[SCXML spec, 3.4.2.](https://www.w3.org/TR/scxml/#parallel)
In the context of children of a <parallel> node, the following is allowed:
> <parallel> Defines a nested set of parallel regions. Occurs 0 or more times.

[XState docs, Final/Parallel states](https://xstate.js.org/docs/guides/final.html#parallel-states)
> When every child state node in a parallel state node is _done_, the parent parallel state node is also _done_. That is, when every final state node in every child compound node is reached, the done(...) event will be raised for the parallel state node.

Reproduction here: https://codesandbox.io/s/damp-shadow-8bvpn?file=/src/the.machine.ts

**Comment by Andarist** [comment]
I've dug into this and it turns out that this isn't a bug - at least not according to the SCXML spec.

The problem is that parallel states are not compound states - only `<state>` elements can be classified as compound. And the only moment where `done.state.id` events are created is in the [`enterStates` procedure](https://www.w3.org/TR/scxml/#enterStates). As we may see there the `done.state.id` event can only be created for the parent and grandparent of the final state, in your example, there is `onDone` defined on the grandgrandparent as well and this is the one that isn't selected.

I wouldn't necessarily call this behavior intuitive so maybe there is room for improvement here - it would also mean deviating from the SCXML semantics (which I'm fine with). Thoughts, @davidkpiano ?

**Comment by davidkpiano** [comment]
I agree, this could be an additive enhancement; one that's not contradictory to SCXML's normal behavior (except in unlikely/theoretical edge-cases) so I would be open to deviating slightly from the spec here to support this.

**Comment by johnnydecimal** [comment]
Thanks gents. Alternatively if you’d prefer to stay on-spec, I’d be happy to contribute docs explaining an alternate solution (after I’ve figured out what that would be!).

**Comment by Andarist** [comment]
@johnnydecimal I'm looking into implementing this - should have a PR for it this week, I only need to write down tests for this and rethink how this plays with some other parts of the algorithm that we already have in place.
